src/comb_filter.rs

```diff
@@ -21,26 +33,129 @@ pub enum Error {
+        for buffer in self.buffers.iter_mut(){
+            buffer.reset();
+            buffer.set_read_index(index);
+        }
     }
 
     pub fn process(&mut self, input: &[&[f32]], output: &mut [&mut [f32]]) {
-        todo!("implement");
+        //todo!("implement");
+        for  (out_chunk,in_chunk) in output.iter_mut().zip(input)
+        {   
+            for (channel_id,(outVal, inVal)) in out_chunk.iter_mut().zip(in_chunk.iter()).enumerate()
+            {
+                self.calculate_filter(inVal, outVal, channel_id);
+            }
+        }
     }
 
     pub fn set_param(&mut self, param: FilterParam, value: f32) -> Result<(), Error> {
-        todo!("implement")
+        //todo!("implement")
+        use FilterParam::*;
+        match param
+        {
+            Delay => {
+                match self.filterType {
+                    FilterType::FIR => {
+                        if value > self.max_delay || value < 0.0{
+                            println!("Delay must be greater than 0 and less than or equal to max delay size!");
+                            return Err(Error::InvalidValue { param, value });
+                        }
+                    },
+                    FilterType::IIR => {
+                        if value <= 0.0 {
+                            println!("Delay cannot be zero or less than zero in an IIR filter!");
+                            return Err(Error::InvalidValue { param, value });
+                        }
+                        if value > self.max_delay{
+                            println!("Delay must be less than or equal to max delay size!");
+                            return Err(Error::InvalidValue { param, value });
+                        }
+                    }
+                }
+                self.delay = value;
+                let index = self.find_read_index((value*self.sample_rate) as i32);
+                for buffer in self.buffers.iter_mut(){
+                    buffer.set_read_index(index);
+                }
+            },
+            Gain => {
+                if value < 0.0 || value > 1.0{
+                    println!("Gain must be between 0.0 and 1.0!");
+                    return Err(Error::InvalidValue { param, value });
+                }
+                self.gain = value
+            },
+            _ => {
+                println!("Invalid Parameter!");
+                return Err(Error::InvalidValue { param, value });
+            }
+        }
+        Ok(())
     }
 
     pub fn get_param(&self, param: FilterParam) -> f32 {
-        todo!("implement")
+        //todo!("implement");
+        use FilterParam::*;
+        match param
+        {
+            Delay => return self.delay,
+            Gain => return self.gain
+        }
+    }
+    fn find_read_index(&mut self, del_in_samps: i32) -> usize {
+        let write_index = self.buffers[0].get_write_index() as i32;
```

Comment:
Ensure that the usage of self.buffers[0] is appropriate for your design. If it also work with any other channels, you might also want to include them.